cmd/mdatagen/templates/component_test.go.tmpl

```diff
@@ -33,27 +32,6 @@ import (
 {{ end }}
 )
 
-// assertNoErrorHost implements a component.Host that asserts that there were no errors.
```

Comment:
Should we instead generate an assert no error StatusReporter?

Reply:
I'd do it in phases. I am fairly certain that if we generate an assert no error StatusReporter, tests will fail because it's ok for them to report status errors on start if say we have no database to connect to.

I would also create this reporter in a common lib rather than in every generated test.

Reply:
I think we can assert if the status is fatal not on error.